arwen/contexts/metering.go

```diff
@@ -82,6 +145,29 @@ func (context *meteringContext) GasLeft() uint64 {
 	return gasProvided - gasUsed
 }
 
+// GasForwarded returns the amount of gas used by the current contract for the
+// execution of other contracts
+func (context *meteringContext) GasForwarded() uint64 {
+	return context.gasForwarded
+}
+
+// ForwardGas accumulates the gas forwarded by the current contract for the execution of other contracts
+func (context *meteringContext) ForwardGas(gas uint64) {
+	context.gasForwarded = math.AddUint64(context.gasForwarded, gas)
+}
+
+func (context *meteringContext) GasUsedByContract() uint64 {
+	runtime := context.host.Runtime()
+	compilationGas := math.SubUint64(context.initialGasProvided, runtime.GetVMInput().GasProvided)
```

Comment:
having code like this makes it easy to use it wrongly. The same issue when the current execution / compilation cost is effectively lost. version 0.4, from 0.5 onwards this was corrected.

Reply:
there is no instance stacking in the mainnet currently.

Reply:
compilation / AoT prepare gas could be save in a separate field. we have to be carefull about backward compatibility

Reply:
Will make a separate field.